src/pages/waypoint/tutorials/[...tutorialSlug]/index.tsx

```diff
@@ -0,0 +1,32 @@
+import { ProductOption } from 'lib/learn-client/types'
+import TutorialView from 'views/tutorial-view'
+import {
+  getTutorialPagePaths,
+  getTutorialPageProps,
+  TutorialPageProps,
+  TutorialPagePaths,
+} from 'views/tutorial-view/server'
+
+export default function TutorialPage({
+  tutorial,
+}: TutorialPageProps): React.ReactElement {
+  return <TutorialView {...tutorial} />
+}
+
+export async function getStaticProps({
+  params,
+}): Promise<{ props: TutorialPageProps }> {
+  const props = await getTutorialPageProps(
+    ProductOption['waypoint'],
+    params.tutorialSlug
+  )
+  return props
```

Comment:
If you want to get the `ProductSwitcher` working in this PR for this page, the props object returned from `getStaticProps` will need to include a `product` property that is a `Product` object (see `types/products.ts`. This is something we have to do for every page component.

The Waypoint downloads page (`src/pages/waypoint/downloads/index.tsx`) is a recent example:

- imports the JSON data from `src/data/waypoint.json`
- creates a `product` variable in `getStaticProps` and type casts `waypointData` as `Product`
- sends the `product` object to `generateStaticProps` needed for that page
- `generateStaticProps` does it's logic and then returns the `product` object under the `props` property

I think your options in this file after importing `waypoint.json` would be:

- update `getTutorialPageProps` to accept a `Product` object and to return the object in `props`
- update this line in `getStaticProps` to be:
  ```ts
  return {
    ...props,
    product
  }
  ```

Reply:
Cool! I'll check out the product object. ty 😄